src/tls_stream.rs

```diff
@@ -501,6 +501,30 @@ impl<S> TlsStream<S>
         };
 
         unsafe {
+            // check if we trust the root-CA explicitly
+            let mut para_flags = winapi::CERT_CHAIN_POLICY_IGNORE_ALL_REV_UNKNOWN_FLAGS;
+            if let Some(ref mut store) = self.cert_store {
+                let cert_chain = *cert_chain.0;
+                if cert_chain.cChain > 0 {
+                    let first_rgp_chain = **cert_chain.rgpChain;
+                    if first_rgp_chain.cElement > 0 {
+                        let elements = slice::from_raw_parts(
+                            first_rgp_chain.rgpElement as *mut &mut winapi::CERT_CHAIN_ELEMENT,
+                            first_rgp_chain.cElement as usize);
+                        let final_element = elements.last().unwrap();
+                        
+                        let root_cert = CertContext::from_inner(final_element.pCertContext);
+                        for cert in store.certs() {
+                            if try!(cert.is_equal(&root_cert)) {
```

Comment:
Ouch.
The current revision contains an implementation, which doesn't
need 2 mem::forget's and might be the most elegant solution 
(without using a RAII-"FreeGuard", that requires more boilerplate).

A simple `match` would require 2 mem::forget's, so I slightly prefer this one.
Are there any disadvantages I didn't consider there? 
(except maybe a tiny performance impact?)